src/main/java/com/epam/edumanagementsystem/admin/timetable/rest/api/TimetableController.java

```diff
@@ -54,23 +55,23 @@ public String openingTimetablePage(@PathVariable("academicClassName") String aca
 
         if (timetableService.findTimetableByAcademicClassId(academicClass.getId()) != null) {
             if (timetableService.findTimetableByAcademicClassId(academicClass.getId()).getStatus().equalsIgnoreCase("Edit") &&
-                    coursesService.getCoursesWithEditStatusByAcademicCourseId(academicClass.getId()).size() != 0 &&
-                    coursesService.getCoursesWithActiveStatusByAcademicCourseId(academicClass.getId()).size() != 0) {
-                List<CoursesForTimetable> activeCourses = coursesService.getCoursesWithActiveStatusByAcademicCourseId(academicClass.getId());
-                List<CoursesForTimetable> editCourses = coursesService.getCoursesWithEditStatusByAcademicCourseId(academicClass.getId());
+                    !coursesService.getCoursesWithEditStatusByAcademicClassId(academicClass.getId()).isEmpty() &&
+                    !coursesService.getCoursesWithActiveStatusByAcademicClassId(academicClass.getId()).isEmpty()) {
+                List<CoursesForTimetable> activeCourses = coursesService.getCoursesWithActiveStatusByAcademicClassId(academicClass.getId());
+                List<CoursesForTimetable> editCourses = coursesService.getCoursesWithEditStatusByAcademicClassId(academicClass.getId());
                 for (CoursesForTimetable activeCourse : activeCourses) {
```

Comment:
Possibly each for loop within the controller class could be transformed to a separate method, also naming for coursesForTimetable class could be changed to "lesson", which itself would shorten names of many methods. There aslo could be a date validating method instead of long if/elses. Maybe each if case in methods could  act as a separate method.